lib/fluent/plugin/output.rb

```diff
@@ -185,13 +185,25 @@ def initialize
         @simple_chunking = nil
         @chunk_keys = @chunk_key_time = @chunk_key_tag = nil
         @flush_mode = nil
+        @timekey_zone = nil
       end
 
       def acts_as_secondary(primary)
         @as_secondary = true
         @primary_instance = primary
+        @chunk_keys = @primary_instance.chunk_keys || []
+        @chunk_key_tag = @primary_instance.chunk_key_tag || false
+        if @primary_instance.chunk_key_time
+          @chunk_key_time = @primary_instance.chunk_key_time
+          @timekey_zone = @primary_instance.timekey_zone
+          @output_time_formatter_cache = {}
+        end
+
+        require_override = !self.class.instance_methods(false).include?(:extract_placeholders)
         (class << self; self; end).module_eval do
-          define_method(:extract_placeholders){ |str, metadata| @primary_instance.extract_placeholders(str, metadata) }
```

Comment:
Overwriting this method is to use `@chunk_keys` and some other instance variables of primary plugin instances.
But your update brings these instance variable and these values into secondary plugin instances too.
So I think we can just remove this L194.